src/caret_analyze/infra/lttng/lttng.py

```diff
@@ -370,79 +412,102 @@ def _parse_lttng_data(
+            event_remapper = IDRemapperCollection()
+            for trace_dir in trace_dir_or_events:
+                event_collection = EventCollection(
+                    trace_dir, force_conversion)  # type: ignore
+                print('{} events found.'.format(len(event_collection)))
+
+                common = LttngEventFilter.Common()
+                begin, end = event_collection.time_range()
+                common.start_time, common.end_time = begin, end
+
+                # Offset is obtained for conversion from
+                # the monotonic clock time to the system time.
+                for event in event_collection:
+                    event_name = event[LttngEventFilter.NAME]
+                    if event_name == 'ros2_caret:caret_init':
+                        offset = Ros2Handler.get_monotonic_to_system_offset(event)
+                        break
+
+                handler = Ros2Handler(data, event_remapper, offset)
+
+                init_events = []
+                run_events = []
+                filtered_event_count = 0
+
+                # distribute all trace events to init_events and run_events
+                init_event_names = set(Lttng._prioritized_init_events)
+                for event in event_collection:
+                    event_name = event[LttngEventFilter.NAME]
+                    if len(event_filters) > 0 and \
+                            any(not f.accept(event, common) for f in event_filters):
+                        continue
+                    if event_name in init_event_names:
+                        init_events.append(event)
+                    else:
+                        run_events.append(event)
+                    filtered_event_count += 1
+
+                Lttng.apply_init_timestamp(init_events, offset)
+
+                import functools
+                init_events.sort(key=functools.cmp_to_key(Lttng._compare_init_event))
+                handler.create_init_handler_map()
+                for event in tqdm(
+                        iter(init_events),
+                        total=len(init_events),
+                        desc='loading',
+                        mininterval=1.0):
+                    tid_remapper.remap(event)
+                    pid_remapper.remap(event)
+                    event_name = event[LttngEventFilter.NAME]
+                    handler_ = handler.handler_map[event_name]
+                    handler_(event)
+
+                handler.create_runtime_handler_map()
+                for event in tqdm(
+                        iter(run_events),
+                        total=len(run_events),
+                        desc='loading',
+                        mininterval=1.0):
+                    if store_events:
+                        event_dict = {
+                            k: get_field(event, k) for k in event
+                        }
+                        events.append(event_dict)
+                    tid_remapper.remap(event)
```

Comment:
Please revert `filtered_event_count += 1`. It is necessary to count filtered runtime events.

```suggestion
                        events.append(event_dict)
                    filtered_event_count += 1
                    tid_remapper.remap(event)
```

Reply:
Sorry, I have been confusing. Current code is correct.